lib/internal/buffer.js

```diff
@@ -458,6 +529,66 @@ function readDoubleForwards(offset) {
 }
 
 // Write integers.
+function writeBigU_Int64LE(buf, value, offset, min, max) {
+  // TODO: perform any conversions on value?
+  checkInt(value, min, max, buf, offset, 7);
+
+  buf[offset++] = Number(value & 0xffn);
+  value = value >> 8n;
+  buf[offset++] = Number(value & 0xffn);
+  value = value >> 8n;
+  buf[offset++] = Number(value & 0xffn);
+  value = value >> 8n;
+  buf[offset++] = Number(value & 0xffn);
+  value = value >> 8n;
+  buf[offset++] = Number(value & 0xffn);
+  value = value >> 8n;
+  buf[offset++] = Number(value & 0xffn);
+  value = value >> 8n;
+  buf[offset++] = Number(value & 0xffn);
+  value = value >> 8n;
+  buf[offset++] = Number(value & 0xffn);
+  return offset;
```

Comment:
The way this is handled in the other functions is actually a performance optimization. It is not clear if that applies to BigInt as well. I actually guess it will not.

Reply:
Which performance optimization do you mean?

Reply:
It is faster to calculate the next number by always only shifting by 8 instead of shifting by 8, 16, 24 ...

Reply:
I don't think shifting by 8, 16, 24 would be more readable. Unless it's faster in V8 6.7, I'd prefer to keep the same approach as the other functions.

Reply:
It makes it slower:

```
                                                                  confidence improvement accuracy (*)    (**)   (***)
 buffers\\buffer-write.js n=1000000 type='BigInt64BE' buffer='fast'         ***    -12.89 %      Â±1.48% Â±1.97% Â±2.56%
 buffers\\buffer-write.js n=1000000 type='BigInt64BE' buffer='slow'         ***    -13.37 %      Â±1.69% Â±2.25% Â±2.93%
 buffers\\buffer-write.js n=1000000 type='BigInt64LE' buffer='fast'         ***    -12.28 %      Â±1.59% Â±2.12% Â±2.76%
 buffers\\buffer-write.js n=1000000 type='BigInt64LE' buffer='slow'         ***    -14.50 %      Â±1.93% Â±2.57% Â±3.35%
 buffers\\buffer-write.js n=1000000 type='BigUInt64BE' buffer='fast'        ***    -14.91 %      Â±2.04% Â±2.71% Â±3.53%
 buffers\\buffer-write.js n=1000000 type='BigUInt64BE' buffer='slow'        ***    -13.10 %      Â±1.70% Â±2.26% Â±2.94%
 buffers\\buffer-write.js n=1000000 type='BigUInt64LE' buffer='fast'        ***    -13.06 %      Â±1.65% Â±2.20% Â±2.86%
 buffers\\buffer-write.js n=1000000 type='BigUInt64LE' buffer='slow'        ***    -13.59 %      Â±2.05% Â±2.73% Â±3.55%
```

So leaving as-is.

Reply:
Just curious, did you check what happens when you coerce the low and high half to Number and shift/assign those?

I believe BigInts are still boxed all the time and their shift operator is a runtime call so you can probably eke out better performance if you remove them from the equation first.

Reply:
```
                                                                     confidence improvement accuracy (*)    (**)    (***)
 buffers\\buffer-write.js n=1000000 type='BigInt64BE' buffer='fast'         ***    137.44 %      Â±6.21% Â±8.35% Â±11.03%
 buffers\\buffer-write.js n=1000000 type='BigInt64BE' buffer='slow'         ***    139.98 %      Â±3.15% Â±4.22%  Â±5.53%
 buffers\\buffer-write.js n=1000000 type='BigInt64LE' buffer='fast'         ***    138.94 %      Â±4.38% Â±5.88%  Â±7.77%
 buffers\\buffer-write.js n=1000000 type='BigInt64LE' buffer='slow'         ***    140.36 %      Â±3.38% Â±4.53%  Â±5.97%
 buffers\\buffer-write.js n=1000000 type='BigUInt64BE' buffer='fast'        ***    149.35 %      Â±6.68% Â±8.98% Â±11.86%
 buffers\\buffer-write.js n=1000000 type='BigUInt64BE' buffer='slow'        ***    151.90 %      Â±3.46% Â±4.63%  Â±6.07%
 buffers\\buffer-write.js n=1000000 type='BigUInt64LE' buffer='fast'        ***    153.97 %      Â±5.24% Â±7.03%  Â±9.27%
 buffers\\buffer-write.js n=1000000 type='BigUInt64LE' buffer='slow'        ***    150.56 %      Â±3.07% Â±4.13%  Â±5.45%
```